DfE.Hexagonal.Common/Validation/ValidationResult.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DfE.Hexagonal.Common.Validation
+{
+    /// <summary>
+    /// contains a validation result and a dictionary of errors, if applicable
+    /// </summary>
+    public class ValidationResult
+    {
+        /// <summary>
+        /// is the thing valid?
+        /// </summary>
+        public bool IsValid { get; }
+
+        /// <summary>
+        /// key/error pairs for each member of an IValidatable
+        /// </summary>
+        public Dictionary<string, string> Errors { get; }
+
+        public ValidationResult(bool isValid, Dictionary<string, string> errors)
+        {
+            IsValid = isValid;
+            Errors = errors;
+        }
```

Comment:
The API for this constructor currently forces the user to ALWAYS supply a list of errors, even if no errors exist.

As a bit of sweet syntactic sugar, if we turn it the dictionary into an optional parameter, this allows folks to optionally use the shorter form of `new ValidationResult(true)`

```csharp
        public ValidationResult(bool isValid, Dictionary<string, string>? errors = null)
        {
            IsValid = isValid;
            Errors = errors ?? new();
        }
```